eng/pipelines/templates/scripts/validate-changes-on-allowed-path.ps1

```diff
@@ -0,0 +1,55 @@
+<#
+.SYNOPSIS
+
+.DESCRIPTION
+The script is to validate if the diff changes between base and target are within the scope of allowlist.
+
+.PARAMETER BaseBranch
+The base branch where we want to merge changes from. E.g. main branch in docs repo
+
+.PARAMETER TargetBranch
+The target branch where we want to merge changes to. E.g. live branch in docs repo
+
+.PARAMETER AllowlistPath
+The file path of allowlist.
+#>
+param(
+  [Parameter(mandatory=$true)]
+  [string] $BaseBranch,
+  [Parameter(mandatory=$true)]
+  [string] $TargetBranch,
+  [Parameter(mandatory=$true)]
+  [string] $AllowlistPath
+)
+
+# Working branch is base branch, e.g. main
+# Fetch all branches, e.g. main
+Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" fetch origin"
+git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" fetch origin | Write-Host
+
+# Live branch tracks remote live branch.
+Write-Host "git checkout -b $BaseBranch -t origin/$BaseBranch"
+git rev-parse --verify $BaseBranch
+if ($LASTEXITCODE -eq 0) {
+  git checkout $BaseBranch | Write-Host
+}
+else {
+  git checkout -b $BaseBranch -t origin/$BaseBranch | Write-Host
+}
+
+# git diff on all files
+$changedFiles = git diff --name-only $TargetBranch
```

Comment:
Lets reuse https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/get-changedfiles.ps1 so that we have a common set of code we use for getting the list of changed files which already test via usage in other parts.

Reply:
Though it is simple line, I think reuse can help us deal with error cases in a standard way.